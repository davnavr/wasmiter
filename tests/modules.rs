#[test]
fn basic_module() {
    let wat = r#"(module
    (func (export "add_five") (param i32) (result i32)
        local.get 0
        i32.const 5
        i32.add))
"#;
    let wasm = wat::parse_str(wat).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn exception_handling() {
    let wasm = wat::parse_str(include_str!("modules/exception_handling.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn all_the_things() {
    let wasm = wat::parse_str(include_str!("modules/all_the_things.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}
