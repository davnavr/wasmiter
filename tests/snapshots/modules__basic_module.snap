---
source: tests/modules.rs
expression: "wasmiter::parse_module_sections(wasm.as_slice()).unwrap().display_module()"
---
(module
(type (; 0 ;) (func (param i32) (result i32)))

;; function section count = 1
(export "add_five" (func 0))

(func (type 0) ;; code size = 7
  local.get 0
  i32.const 0x00000005 (; 5 signed, 5 unsigned ;)
  i32.add
)
)
